group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetricsShard.java

```diff
@@ -145,17 +133,14 @@ public void incrementNumOffsets() {
     }
 
     /**
-     * Increment the number of consumer groups.
+     * Set the number of consumer groups. The method is the only way to update
+     * the map and is called by the scheduled task that updates the metrics
+     * in {@link org.apache.kafka.coordinator.group.GroupCoordinatorShard}.
```

Comment:
nit: "This method should be the only way.." and add "..breaking this will result in inconsistent behavior" as well as for `setClassicGroupGauges`